draft-ietf-quic-transport.md

```diff
@@ -1641,6 +1641,14 @@ of connection establishment.  By giving the client a different connection ID to
 use, a server can cause the connection to be routed to a server instance with
 more resources available for new connections.
 
+If a server receives a client Initial with a unverifiable Retry token,
+it knows the client will not accept another Retry token.  It can either
+proceed with the handshake without verifying the token or immediately close
```

Comment:
I don't think that we should allow the server to continue.  Why not make the INVALID_TOKEN mandatory?

Reply:
If we are to make this a mandatory requirement, under the assumption that the servers would act like that, should we also suggest that a server should issue tokens in a way that a Retry token and a NEW_TOKEN token are distinguishable from each other?

Unless we recommend servers building a token in that way, I do not think making this a MUST has benefits.

Reply:
If we assume that servers only require Retries when they’re under attack, they might not bother checking the validity of a token when they’re not under attack.

Reply:
> A token SHOULD be constructed in a way that allows the server to distinguish it from tokens that are sent in Retry packets as they are carried in the same field.

-- [Section 8.1.2](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#validate-future)

We could strengthen that to MUST if you like.  It probably should be anyway.

Reply:
The server could decide it's not under heavy load and proceed with the handshake, even if the token doesn't constitute address validation.  ie: The token could expire on a shorter timeline than the session ticket/etc, so 0-RTT could succeed, but address validation would fail.

Reply:
I'm persuaded by Marten's comment here (which I missed at first, I think because we commented at almost the same time).

Reply:
I still think the requirement to make them distinguishable should be a MUST.  Otherwise, a change of token generation method could make an old NEW_TOKEN token appear to be an invalid Retry token.